gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/task/DagProcessingEngineMetrics.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.service.modules.orchestration.task;
+
+import java.util.concurrent.ConcurrentHashMap;
+
+import java.util.concurrent.ConcurrentMap;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.gobblin.metrics.ContextAwareMeter;
+import org.apache.gobblin.metrics.MetricContext;
+import org.apache.gobblin.service.modules.orchestration.DagActionStore;
+import org.apache.gobblin.metrics.ServiceMetricNames;
+
+@Slf4j
+/**
+ * Used to track all metrics relating to processing dagActions when DagProcessingEngine is enabled. The metrics can be
+ * used to trace the number of dagActions (which can be further broken down by time) at various points of the system,
+ * starting from addition to the DagActionStore to observation in the DagActionChangeMonitor through the
+ * DagProcessingEngine pipeline (DagManagement -> DagTaskStreamImpl -> MySqlMultiActiveLeaseArbiter -> DagProc).
+ */
+public class DagProcessingEngineMetrics {
+  MetricContext metricContext;
+  // Declare map of dagActionType to a ContextAwareMeter for each metric
+  private ConcurrentMap<String, ContextAwareMeter> dagActionsStoredMeters = new ConcurrentHashMap();
+  private ConcurrentMap<String, ContextAwareMeter>dagActionsObservedMeters =  new ConcurrentHashMap();
+  private ConcurrentMap<String, ContextAwareMeter>dagActionsLeasesObtainedMeters =  new ConcurrentHashMap();
```

Comment:
nit: all of these are missing a space between `>` and the var name.  there are also too many spaces between `= new`.

finally, naming them `XYZMeters` sounds like they're a list/set/sequence collection, rather than a mapping.

hence, suggest: `leaseObtainedMeterByDagActionType` or `initializeSucceededMeterByDagActionType`